lib/pbench/server/api/resources/__init__.py

```diff
@@ -1555,6 +1567,24 @@ def _dispatch(
         else:
             params = ApiParams(None, None, None)
 
+        audit = None
+        if schema.operation is not OperationCode.READ:
```

Comment:
I considered it and rejected the idea since we're not auditing reads. In any case, it would require re-working the authorization code so the caller knows. If we decide to be that paranoid, it can be added later.

Reply:
> I considered it and rejected the idea since we're not auditing reads.

I'm confused:  aren't we not auditing reads simply because of line 1571 here?  That is, if we added a condition which audited reads-by-admin here, then we would then be auditing (_some_) reads...are you saying that there is more required to make auditing reads (in general) work?

> it would require re-working the authorization code so the caller knows.

If we just changed this `if` statement here, then it would result in auditing _all_ reads by an ADMIN.  Are you saying that the rework would be required in order to be able to avoid auditing an ADMIN reading his/her own stuff?

Reply:
If user `dbutenhof` has `ADMIN` role I don't want to audit every time I read one of my own datasets. This might be different if we had only a traditional `admin` *user*, but the concept of roles complicates this. We would want to audit (if at all) only security *exceptions* rather than *all* reads.

*That* would require hooking into the authorization internals to report that the check succeeded only because of the `ADMIN` role. I'm not convinced that's even what we want, but I'm pretty certain we don't want to record "every read" from an `ADMIN` user.

Reply:
> I'm pretty certain we don't want to record "every read" from an ADMIN user.

Agreed.